File: src/codes/Users_Manual/link/visualizer_calc.cpp
//~#include<iostream>
//~#include<autd3.hpp>
//~#include "autd3/link/visualizer.hpp"
//~int main() {
auto autd = autd3::ControllerBuilder()
                .add_device(autd3::AUTD3(autd3::Vector3::Zero()))
                .open(autd3::link::Visualizer::builder());

autd3::Vector3 center = autd.geometry().center() + autd3::Vector3(0, 0, 150);
autd3::gain::Focus g(center);
autd.send(g);

std::vector points{center};
const auto p =
    autd.link().calc_field(points, autd.geometry(), autd3::Segment::S0, 0);
std::cout << "Acoustic pressure at (" << center.x() << ", " << center.y()
          << ", " << center.z() << ") = " << p[0] << std::endl;
//~return 0; }

File: src/codes/Users_Manual/link/visualizer_calc.rs
# use autd3::prelude::*;
# use autd3_link_visualizer::Visualizer;
# #[allow(unused_variables)]
# #[tokio::main]
# async fn main() -> Result<(), Box<dyn std::error::Error>> {
let mut autd = Controller::builder([AUTD3::new(Vector3::zeros())])
    .open(Visualizer::builder()).await?;

let center = autd.geometry.center() + Vector3::new(0., 0., 150.0 * mm);

autd.send(Focus::new(center)).await?;

let p = autd.link.calc_field(&[center], &autd.geometry, Segment::S0, 0)?;
println!(
    "Acoustic pressure at ({}, {}, {}) = {}",
    center.x, center.y, center.z, p[0]
);

# autd.close().await?;
# Ok(())
# }
